### PR TITLE
Added boolean to return security groups as list (rather than scalar)

### PR DESCRIPTION
**Pull request by Hefeweizen**
##### SUMMARY
I need to iterate over the list of security groups.  Rather than parsing locally (every time I consume them), I modified ec2.py to return a list.

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
contrib/inventory/ec2.ini
contrib/inventory/ec2.py

##### ANSIBLE VERSION
2.3

### CODE REVIEW

File: contrib/inventory/ec2.py [hunk 1]
```diff
@@ -1079,8 +1082,12 @@ def get_host_info_dict_from_instance(self, instance):
                 for group in value:
                     group_ids.append(group.id)
                     group_names.append(group.name)
-                instance_vars["ec2_security_group_ids"] = ','.join([str(i) for i in group_ids])
-                instance_vars["ec2_security_group_names"] = ','.join([str(i) for i in group_names])
+                if self.security_groups_as_list:
+                    instance_vars["ec2_security_group_ids"] = json.dumps([str(i) for i in group_ids])
```
Comment (willthames) [inline contrib/inventory/ec2.py:1086]: I think you can get away with `str(group_ids)` here:

```
>>> a = ["sg-123", "sg-abc"]
>>> str(a)
"['sg-123', 'sg-abc']"
>>> import json
>>> json.dumps(a)
'["sg-123", "sg-abc"]'
```

Comment (willthames) [reply contrib/inventory/ec2.py:1086]: Either way, the list comprehension seems unnecessary

File: contrib/inventory/ec2.py [hunk 2]
```diff
@@ -1079,8 +1082,12 @@ def get_host_info_dict_from_instance(self, instance):
                 for group in value:
                     group_ids.append(group.id)
                     group_names.append(group.name)
-                instance_vars["ec2_security_group_ids"] = ','.join([str(i) for i in group_ids])
```
Comment (willthames) [inline contrib/inventory/ec2.py:1082]: No list comprehension needed here either

```
','.join(group_ids)
```

should work fine

**Comment by s-hertel** [comment]
Hi, @Hefeweizen. We are not working enhancing the inventory scripts since adding the inventory plugins (new in Ansible 2.4). This is a feature you can already do with the [aws_ec2 inventory plugin](https://docs.ansible.com/ansible/latest/plugins/inventory/aws_ec2.html) using the constructed option `compose`. Feel free to reach out if you have questions.

**Pull request closed without merging**